src/wsrep_provider_v26.cpp

```diff
@@ -582,6 +606,19 @@ wsrep::wsrep_provider_v26::wsrep_provider_v26(
     {
         throw wsrep::runtime_error("Failed to initialize wsrep provider");
     }
+
+    if (encryption_enabled)
+    {
+        const std::vector<unsigned char>& key = server_state.get_encryption_key();
+        if (key.size())
+        {
+            wsrep::const_buffer const_key(key.data(), key.size());
+            if(enc_set_key(const_key))
+            {
+                throw wsrep::runtime_error("Failed to set encryption key");
+            }
+        }
```

Comment:
Should the case `encryption_enabled` and no encryption key set be considered an error here?

Reply:
Good question, but i think not. Maybe key is not set at that point and will be provided later. @ayurchen thoughts ?

Reply:
Well, 0-sized key is technically a key. But seriously, wsrep API allows setting a key at any time between provider initialization and `connect()` call. (The reason for it was to have a single way to set the key)
So here, I think it boils down to a decision:
 * freak out right away OR
 * freak out before `connect()` call if the key is still 0

The upside of the second approach is greater application initialization flexibility. The downside is that we may freak out much later in the process. And while this does not change wsrep-lib signatures, we must decide on it now as it defines the order of initialization calls.